packages/plugins/response-cache-cloudflare-kv/src/invalidate.ts

```diff
@@ -0,0 +1,44 @@
+import type { CacheEntityRecord } from '@envelop/response-cache';
+import type { KvCacheConfig } from '.';
+import { buildEntityKey } from './cache-key';
+
+export async function invalidate(
+  entities: Iterable<CacheEntityRecord>,
+  config: KvCacheConfig,
+): Promise<void> {
+  const kvPromises: Promise<unknown>[] = []; // Collecting all the KV operations so we can await them all at once
+
+  for (const entity of entities) {
+    const entityKey = buildEntityKey(entity.typename, entity.id, config.keyPrefix);
+
+    for await (const kvKey of _getAllKvKeysForPrefix(entityKey, config)) {
+      if (kvKey.metadata?.operationKey) {
+        kvPromises.push(config.KV.delete(kvKey.metadata?.operationKey));
+        kvPromises.push(config.KV.delete(kvKey.name));
+      }
+    }
+  }
+
+  await Promise.allSettled(kvPromises);
+}
+
+export async function* _getAllKvKeysForPrefix(prefix: string, config: KvCacheConfig) {
+  let keyListComplete = false;
+  let cursor: string | undefined;
+
+  do {
+    const kvListResponse = await config.KV.list<{ operationKey: string }>({
+      prefix,
+      cursor,
+    });
+    keyListComplete = kvListResponse.list_complete;
+
+    if (!kvListResponse.list_complete) {
+      cursor = kvListResponse.cursor;
+    }
+
+    for (const keyResult of kvListResponse.keys) {
+      yield keyResult;
+    }
+  } while (!keyListComplete);
+}
```

Comment:
Perhaps you can use a for loop here ?

```suggestion
  for(let kvListResponse: ResponseType | undefined ; kvListResponse?.list_complete;) {
    const kvListResponse = await config.KV.list<{ operationKey: string }>({
      prefix,
      cursor: kvListResponse.cursor,
    });
    
    for (const keyResult of kvListResponse.keys) {
      yield keyResult;
    }
  }
}
```

Reply:
I personally find the `do ... while` more readable. But happy to switch this if you feel strongly about it.

Reply:
If you think it's more readable, let it that way :-)